api/python/cellxgene_census/src/cellxgene_census/experimental/diffexp/memento/diff_expr.py

```diff
@@ -61,15 +94,23 @@ def drop_invalid_data(estimators_df: pl.DataFrame) -> pl.DataFrame:
 
 
 def compute_all(
-    cube_path: str, query_filter: str, treatment: str, n_processes: int, n_features: Optional[int] = None
+    cube_path: str,
+    query_filter: str,
+    treatment: str,
+    n_processes: int,
+    covariates: Optional[List[str]] = ["dataset_id"],
 ) -> Tuple[pd.DataFrame, pstats.Stats]:
     with tiledb.open(os.path.join(cube_path, OBS_GROUPS_ARRAY), "r") as obs_groups_array:
         obs_groups_df = obs_groups_array.query(cond=query_filter or None).df[:]
+        if covariates:
+            obs_groups_df = obs_groups_df[covariates + [treatment, "obs_group_joinid", "n_obs"]]
+        else:
+            covariates = CUBE_LOGICAL_DIMS_OBS
```

Comment:
nit: Is this more readable perhaps? (I scratched my head over the if/else for a second :) )
```suggestion
        obs_groups_df = obs_groups_array.query(cond=query_filter or None).df[:]
        covariates = covariates or CUBE_LOGICAL_DIMS_OBS
        obs_groups_df = obs_groups_df[covariates + [treatment, "obs_group_joinid", "n_obs"]]
```